tests/test_main.py

```diff
@@ -0,0 +1,371 @@
+import json
+
+import pytest
+from asgiref.sync import sync_to_async
+from django.contrib.contenttypes.models import ContentType
+from django.core import management
+from django.urls import reverse
+from django.utils.version import get_version
+from pytest_django.asserts import assertInHTML
+
+from easyaudit.middleware.easyaudit import clear_request, set_current_user
+from easyaudit.models import CRUDEvent, RequestEvent
+from tests.test_app.models import (
+    BigIntForeignKeyModel,
+    BigIntM2MModel,
+    BigIntModel,
+    ForeignKeyModel,
+    M2MModel,
+    Model,
+    UUIDForeignKeyModel,
+    UUIDM2MModel,
+    UUIDModel,
+)
+
+
+@pytest.mark.django_db
+def test_no_migrations(capsys: pytest.CaptureFixture):
+    management.call_command("makemigrations", dry_run=True)
+
+    captured = capsys.readouterr().out
+    assert "No changes detected" in captured
+
+
+def test_no_issues(capsys: pytest.CaptureFixture):
+    management.call_command("check", fail_level="WARNING")
+
+    captured: str = capsys.readouterr().out
+    assert "System check identified no issues" in captured
+
+
+@pytest.mark.parametrize(
+    "model",
+    [
+        BigIntForeignKeyModel,
+        BigIntM2MModel,
+        BigIntModel,
+        ForeignKeyModel,
+        M2MModel,
+        Model,
+        UUIDForeignKeyModel,
+        UUIDM2MModel,
+        UUIDModel,
+    ],
+)
+class TestDjangoCompat:
+    def test_model_state(self, model):
+        """Ensures models have the internal `_state` object."""
+        model_instances = model()
+        assert hasattr(model_instances, "_state")
+
+
+@pytest.mark.django_db
+class TestAuditModels:
+    @pytest.fixture
+    def model(self):
+        return Model
+
+    @pytest.fixture
+    def fk_model(self):
+        return ForeignKeyModel
+
+    @pytest.fixture
+    def m2m_model(self):
+        return M2MModel
+
+    @pytest.fixture
+    def _audit_logger(self, monkeypatch):
+        def _crud(*args, **kwargs):
+            raise ValueError("Test exception")
+
+        monkeypatch.setattr("easyaudit.signals.crud_flows.audit_logger.crud", _crud)
+
+    def test_create_model(self, model):
+        obj = model.objects.create()
+        crud_event_qs = CRUDEvent.objects.filter(
+            object_id=obj.id, content_type=ContentType.objects.get_for_model(obj)
+        )
+        assert crud_event_qs.count() == 1
+
+        crud_event = crud_event_qs.first()
+        data = json.loads(crud_event.object_json_repr)[0]
+        assert data["fields"]["name"] == obj.name
+
+    def test_fk_model(self, model, fk_model):
+        obj = model.objects.create()
+        obj_fk = fk_model(name="test", test_fk=obj)
+        obj_fk.save()
+
+        crud_event = CRUDEvent.objects.filter(
+            object_id=obj_fk.id, content_type=ContentType.objects.get_for_model(obj_fk)
+        ).first()
```

Comment:
Minor change here, this used to be `qs[0]`. `qs.first()` provides an identical result, but uses the built-in Django queryset function instead, which is more "correct".